### PR TITLE
Merge semantic snippets into main

### PR DESCRIPTION
**Pull request by akhera99**
The first two snippets of semantic snippets. 
We want to get feedback on the snippets as they are and catch any early bugs. 

TODO:
- Add the rest of the snippets
- Fix minor issues with spacing that exist in snippets that have blocks
- Put the old snippets under an option and turn them off by default
- Make sure that user created snippets still work

### CODE REVIEW

**Comment by Cosifne** [comment]
Do you have the approval to merge in new feature into 17.3 preview3? By default, P3 is mainly for bug fixing.

**Comment by akhera99** [comment]
> Do you have the approval to merge in new feature into 17.3 preview3? By default, P3 is mainly for bug fixing.

@Cosifne Can we not get this in for 17.3 preview 2 in QB mode?

**Comment by dibarbet** [comment]
> @Cosifne Can we not get this in for 17.3 preview 2 in QB mode?

will probably need QB approval

File: src/EditorFeatures/Core.Wpf/AsyncCompletion/LanguageServerSnippetExpanderAdapter.cs
Comment (jmarolf) [inline src/EditorFeatures/Core.Wpf/AsyncCompletion/LanguageServerSnippetExpanderAdapter.cs:16]: Can we add a comment about why this adapter is necessary? Its unclear to me if we want to keep this type around forever or if this is a temp workaround

Comment (jmarolf) [reply src/EditorFeatures/Core.Wpf/AsyncCompletion/LanguageServerSnippetExpanderAdapter.cs:16]: nevermind I remember Sams answer for this. This is because of layering

**Review by jmarolf — CHANGES_REQUESTED** [review]
@akhera99 it doesn't appear that we have any options to control whether these snippets show up or not? I would assume we would want to merge to main with this feature off by default and only enable it once we can replace the existing `if` and `cw` snippets? Sorry to block this but I think we need some minimal way to enable/disable this feature before we merge to main

**Pull request merged**
